server.go

```diff
@@ -66,6 +85,8 @@ type Server struct {
 	closeListenerDelay time.Duration
 	osExitDelay        time.Duration
 	osExitCode         int
+
+	Uuid func() string
```

Comment:
1) Golang wants you to name this `UUID`
2) Is it necessary that this is a UUID? Why not just `RequestIDFactory`?